edk2toolext/invocables/edk2_ci_build.py

```diff
@@ -75,6 +80,33 @@ def GetPluginSettings(self):
         '''  Implement in subclass to pass dictionary of settings for individual plugins '''
         return {}
 
+    # ####################################################################################### #
+    #                     Verify and Save requested Ci Build Config                           #
+    # ####################################################################################### #
+    def SetToPackage(self, list_of_requested_packages):
+        ''' Confirm the requests package list is valid and configure SettingsManager
+        to build only the requested packages.
+
+        Raise Exception if a requested_package is not supported
+        '''
+        pass
+
+    def SetToArchitecture(self, list_of_requested_architectures):
+        ''' Confirm the requests architecture list is valid and configure SettingsManager
+        to run only the requested architectures.
+
+        Raise Exception if a list_of_requested_architectures is not supported
+        '''
+        pass
+
+    def SetToTarget(self, list_of_requested_target):
```

Comment:
I'm not super clear on what the SetToArchitecture and the SetTo* stuff does/where it's called/etc

Reply:
Ah I see it.

Reply:
So the default implementation is to do nothing- is this desired behavior? Or would we want to throw a not implemented exception

Reply:
SetSettingsToTarget might be better

Reply:
Or should we expect the CI settings to read from the environment?

Reply:
fixed func names.  
Default is to do nothing and not raise.  I think that is choice of implementation
And i think for these 3 settings they are special